spec/models/user_spec.rb

```diff
@@ -166,6 +166,46 @@
       end
     end
 
+    context 'location based' do
+      context 'an anonymous user from a configured location' do
+        let(:user) { User.new(ip_address: 'ip.address2') }
+        let(:rights_xml) do
+          <<-XML
+          <rightsMetadata>
+              <access type="read">
+                <machine>
+                  <location>location1</location>
+                </machine>
+              </access>
+            </rightsMetadata>
+          XML
+        end
+
+        it { is_expected.to be_able_to(:download, file) }
+        it { is_expected.to be_able_to(:download, image) }
+        it { is_expected.to be_able_to(:download, media) }
+      end
+
+      context 'an anonymous user not in the configured location' do
+        let(:user) { User.new(ip_address: 'some.unknown.ip') }
+        let(:rights_xml) do
+          <<-XML
+          <rightsMetadata>
+              <access type="read">
+                <machine>
+                  <location>location1</location>
+                </machine>
+              </access>
+            </rightsMetadata>
+          XML
+        end
+
+        it { is_expected.not_to be_able_to(:download, file) }
+        it { is_expected.not_to be_able_to(:download, image) }
+        it { is_expected.not_to be_able_to(:download, media) }
+      end
+    end
+
```

Comment:
minor quibble:  seems like `rights_xml` is the same for both of these tests.  what about moving the `let(:rights_xml)` out into the wider `context 'location based'` instead of repeating it in each `anonymous user` context?  seems like that might make the test slightly clearer and slightly shorter?